Yoast/javascript
# 187 create tree builder class

## Summary
**Note**: This branch is based on the #119 branch. This should be merged into `develop` before this one can be merged!
 * _[not-user-facing]_ Creates a `TreeBuilder` class. This class chooses the right parser depending on the language in which the source code has been written (e.g. HTML, or Gutenberg in the future).

## Relevant technical choices:

* I changed the folder structure of the `parsedPaper/build` package to pave the way for new formatting languages that will be parsed in the future (e.g. Gutenberg, Markdown).
  * I added a `parsedPaper/build/html` folder, in which I stored the specific functionality for building trees from HTML.
  * My idea: add a folder for each supported language in the future.

## Test instructions

This PR can be tested by following these steps:

* Checkout this branch (``)
* Run `yarn test`.
  * Check if the coverage of `TreeBuilder` is 100%.
* Run the example.
  * Check if the analysis does not give any error in the console and generally runs as expected.

## UI changes

## Quality assurance


Fixes #187
